### PR TITLE
Add git-fixup formula

### PR DESCRIPTION
**Pull request by nunofgs**
Adds a formula for git-fixup. Note that `git-fixup` only returns a success retcode when there are modified files in the staging area.

### CODE REVIEW

File: Library/Formula/git-fixup.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  def test
+    system "git", "init"
+
+    (testpath/"test").write "foo"
+    system "git", "add", "test"
+    system "git", "commit", "--message", "Initial commit"
+
+    IO.write testpath/"test", "bar", mode: "a"
```
Comment (mikemcquaid) [inline Library/Formula/git-fixup.rb]: Needs to be `:mode => "a"` for Ruby 1.8 support.

File: Library/Formula/git-fixup.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class GitFixup < Formula
+  homepage "https://github.com/keis/git-fixup"
+  url "https://github.com/keis/git-fixup/archive/v1.0.0.tar.gz"
+  sha1 "836613c7b9d1ccafaa5f5250b6ff2e125fabf974"
+
+  head "https://github.com/keis/git-fixup.git", :branch => "master"
+
+  def install
+    system "make", "prefix=#{prefix}", "install"
+    zsh_completion.install 'completion.zsh' => '_git-fixup'
```
Comment (mikemcquaid) [inline Library/Formula/git-fixup.rb]: Use double-quotes everywhere.

File: Library/Formula/git-fixup.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  def test
```
Comment (mikemcquaid) [inline Library/Formula/git-fixup.rb]: Use `test do`.

**Comment by nunofgs** [comment]
@mikemcquaid Thank you for the review. I've updated the PR.

File: Library/Formula/git-fixup.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  test do
+    system "git", "init"
+
+    (testpath/"test").write "foo"
+    system "git", "add", "test"
+    system "git", "commit", "--message", "Initial commit"
+
+    IO.write testpath/"test", "bar", :mode => "a"
```
Comment (mikemcquaid) [inline Library/Formula/git-fixup.rb]: There's no `IO.write` in Ruby 1.8. Maybe `File.write`?

**Comment by nunofgs** [comment]
@mikemcquaid As you can probably tell, I'm not very familiar with Ruby :smile:

I've updated the PR. Thanks for the patience.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `git-fixup 1.0.0 (new formula)`.

**Pull request closed without merging**